pilot/pkg/networking/core/v1alpha3/listener_inbound.go

```diff
@@ -0,0 +1,674 @@
+	// clusterName defines the destination cluster for this chain
+	clusterName string
+	// port defines the port configuration for this chain. Note that there is a Port and TargetPort;
+	// most usages should just use TargetPort. Port is mostly used for legacy compatibility and
+	// telemetry.
+	port ServicePort
+	// bind determines where (IP) this filter chain should bind. Note: typically we just end up using
+	// 'virtual' listener and do not literally bind to port; in these cases this just impacts naming
+	// and telemetry.
+	bind string
+
+	// tlsSettings defines the *custom* TLS settings for the chain. mTLS settings are orthogonal; this
+	// only configures TLS overrides.
+	tlsSettings *networking.ServerTLSSettings
+
+	// passthrough should be set to true for the 'passthrough' chains, which are the chains always
+	// present to handle all unmatched traffic. These have a few naming and quirks that require
+	// different configuration.
+	passthrough bool
+
+	// bindToPort determines if this chain should form a real listener that actually binds to a real port,
+	// or if it should just be a filter chain part of the 'virtual inbound' listener.
+	bindToPort bool
+
+	// telemetryMetadata defines additional information about the chain for telemetry purposes.
+	telemetryMetadata util.TelemetryMetadata
+}
+
+// StatPrefix returns the stat prefix for the config
+func (cc inboundChainConfig) StatPrefix() string {
+	if cc.passthrough {
+		// A bit arbitrary, but for backwards compatibility just use the cluster name
+		return cc.clusterName
+	}
+	return "inbound_" + cc.Name(istionetworking.ListenerProtocolHTTP)
+}
+
+// Name determines the name for this chain
+func (cc inboundChainConfig) Name(protocol istionetworking.ListenerProtocol) string {
+	if cc.passthrough {
+		// A bit arbitrary, but for backwards compatibility fixed names are used
+		// Passthrough chains have fix names based on protocol
+		if protocol == istionetworking.ListenerProtocolHTTP {
+			return model.VirtualInboundCatchAllHTTPFilterChainName
+		}
+		return model.VirtualInboundListenerName
+	}
+	// Everything else derived from bind/port
+	return getListenerName(cc.bind, int(cc.port.TargetPort), istionetworking.TransportProtocolTCP)
+}
+
+// ToFilterChainMatch builds the FilterChainMatch for the config
+func (cc inboundChainConfig) ToFilterChainMatch(opt FilterChainMatchOptions) *listener.FilterChainMatch {
+	match := &listener.FilterChainMatch{}
+	match.ApplicationProtocols = opt.ApplicationProtocols
+	match.TransportProtocol = opt.TransportProtocol
+	if cc.passthrough {
+		// Pasthrough listeners do an IP match - but matching all IPs. This is really an IP *version* match,
+		// but Envoy doesn't explicitly have version check.
+		var matchingIP string
+		if cc.clusterName == util.InboundPassthroughClusterIpv4 {
+			matchingIP = "0.0.0.0/0"
+		} else {
+			matchingIP = "::/0"
+		}
+		match.PrefixRanges = []*core.CidrRange{util.ConvertAddressToCidr(matchingIP)}
```

Comment:
may be we should just have pre built ipv6 CidrRange and ipv4 CidrRange and use it directly here instead of building every time?